django/db/models/functions/comparison.py

```diff
@@ -15,6 +16,13 @@ def __init__(self, expression, output_field):
 
     def as_sql(self, compiler, connection, **extra_context):
         extra_context["db_type"] = self.output_field.cast_db_type(connection)
+        if not connection.features.supports_cast_tuple_to_text and any(
```

Comment:
What about if the cast is not to `TextField`? Why `::text` specifically.

Reply:
Then the backend will raise `ProgrammingError`, and at Sarah's suggestion I added a test for this. I'm happy with that, since that matches other invalid casts today.